### ISSUE TITLE
Storage Manager shows that the SMART test on a drive in an expansion unit did not run

### ISSUE THREAD

**Issue by 007revad** [wallacebrf/Synology-SMART-test-scheduler#18]
I used the webui to manually start a short SMART on each drive:
- 1 SSD in the NAS.
- 1 HDD in the NAS.
- 1 HDD in an expansion unit.
- 1 USB HDD in a USB dock.

Storage Manager shows that the SMART test on the drive in an expansion unit did not run: 
![image](https://github.com/user-attachments/assets/74393b61-1366-48b6-8a01-7586a834fb4e)

<br> 

While Storage Manager shows that the SMART tests on the drives in the NAS did run okay:
![image](https://github.com/user-attachments/assets/57bb5934-a98e-4794-8e9d-e5bc99ff57d2)

<br> 

![image](https://github.com/user-attachments/assets/affbf2d1-227d-458f-82eb-d13843feeb4a)

**Comment by wallacebrf** [comment]
Interesting, and confirming this is for short tests only?

To be honest I have not done much testing with short tests beyond if they
Start.

**Comment by wallacebrf** [comment]
What does it look like under the "history" tab?

**Comment by 007revad** [comment]
I haven't tried long tests.

The history tab in storage manager shows:
- 2 healthy quick tests from today for the 2 drives in the NAS.
- The last test for the drive in the expansion unit was on the 10th of Oct.

**Comment by wallacebrf** [comment]
i was able to recreate your issue

started short test on all 9x drives in my DS920+DX517. 

the 4x main drives all show in storage manager that a test occurred, you can see the test dated 11/13
![image](https://github.com/user-attachments/assets/4fb9b765-faad-41cc-b95e-7043ed16e35c)

however none of my expansion unit drives report a test
![image](https://github.com/user-attachments/assets/c92e3c42-2e06-4d0d-8f6d-02e61a201f25)

you can see the last test was on the 10th. 

however smarctl itself does show the results:

```
smartctl 6.5 (build date Sep 26 2022) [x86_64-linux-4.4.302+] (local build)
Copyright (C) 2002-16, Bruce Allen, Christian Franke, www.smartmontools.org

=== START OF INFORMATION SECTION ===
Model Family:     HGST Ultrastar DC HC520 (He12)
Device Model:     HGST HUH721212ALE604
Serial Number:    AAHZW6MH
LU WWN Device Id: 5 000cca 278dc20cb
Firmware Version: LEGNW3D0
User Capacity:    12,000,138,625,024 bytes [12.0 TB]
Sector Sizes:     512 bytes logical, 4096 bytes physical
Rotation Rate:    7200 rpm
Form Factor:      3.5 inches
Device is:        In smartctl database [for details use: -P show]
ATA Version is:   ACS-2, ATA8-ACS T13/1699-D revision 4
SATA Version is:  SATA 3.2, 6.0 Gb/s (current: 6.0 Gb/s)
Local Time is:    Wed Nov 13 17:00:18 2024 CST
SMART support is: Available - device has SMART capability.
SMART support is: Enabled

=== START OF READ SMART DATA SECTION ===
SMART overall-health self-assessment test result: PASSED

General SMART Values:
Offline data collection status:  (0x82) Offline data collection activity
                                        was completed without error.
                                        Auto Offline Data Collection: Enabled.
Self-test execution status:      (   0) The previous self-test routine completed
                                        without error or no self-test has ever
                                        been run.
Total time to complete Offline
data collection:                (   87) seconds.
Offline data collection
capabilities:                    (0x5b) SMART execute Offline immediate.
                                        Auto Offline data collection on/off support.
                                        Suspend Offline collection upon new
                                        command.
                                        Offline surface scan supported.
                                        Self-test supported.
                                        No Conveyance Self-test supported.
                                        Selective Self-test supported.
SMART capabilities:            (0x0003) Saves SMART data before entering
                                        power-saving mode.
                                        Supports SMART auto save timer.
Error logging capability:        (0x01) Error logging supported.
                                        General Purpose Logging supported.
Short self-test routine
recommended polling time:        (   2) minutes.
Extended self-test routine
recommended polling time:        (1281) minutes.
SCT capabilities:              (0x003d) SCT Status supported.
                                        SCT Error Recovery Control supported.
                                        SCT Feature Control supported.
                                        SCT Data Table supported.

SMART Attributes Data Structure revision number: 16
Vendor Specific SMART Attributes with Thresholds:
ID# ATTRIBUTE_NAME                                                   FLAG     VALUE WORST THRESH TYPE      UPDATED  WHEN_FAILED RAW_VALUE
  1 Raw_Read_Error_Rate                                              0x000b   100   100   016    Pre-fail  Always       -       0
  2 Throughput_Performance                                           0x0005   132   132   054    Pre-fail  Offline      -       96
  3 Spin_Up_Time                                                     0x0007   160   160   024    Pre-fail  Always       -       412 (Average 410)
  4 Start_Stop_Count                                                 0x0012   100   100   000    Old_age   Always       -       60
  5 Reallocated_Sector_Ct                                            0x0033   100   100   005    Pre-fail  Always       -       0
  7 Seek_Error_Rate                                                  0x000b   100   100   067    Pre-fail  Always       -       0
  8 Seek_Time_Performance                                            0x0005   128   128   020    Pre-fail  Offline      -       18
  9 Power_On_Hours                                                   0x0012   094   094   000    Old_age   Always       -       46107
 10 Spin_Retry_Count                                                 0x0013   100   100   060    Pre-fail  Always       -       0
 12 Power_Cycle_Count                                                0x0032   100   100   000    Old_age   Always       -       21
 22 Helium_Level                                                     0x0023   100   100   025    Pre-fail  Always       -       100
192 Power-Off_Retract_Count                                          0x0032   099   099   000    Old_age   Always       -       1966
193 Load_Cycle_Count                                                 0x0012   099   099   000    Old_age   Always       -       1966
194 Temperature_Celsius                                              0x0002   214   214   000    Old_age   Always       -       28 (Min/Max 18/44)
196 Reallocated_Event_Count                                          0x0032   100   100   000    Old_age   Always       -       0
197 Current_Pending_Sector                                           0x0022   100   100   000    Old_age   Always       -       0
198 Offline_Uncorrectable                                            0x0008   100   100   000    Old_age   Offline      -       0
199 UDMA_CRC_Error_Count                                             0x000a   200   200   000    Old_age   Always       -       2

SMART Error Log Version: 1
ATA Error Count: 2
        CR = Command Register [HEX]
        FR = Features Register [HEX]
        SC = Sector Count Register [HEX]
        SN = Sector Number Register [HEX]
        CL = Cylinder Low Register [HEX]
        CH = Cylinder High Register [HEX]
        DH = Device/Head Register [HEX]
        DC = Device Command Register [HEX]
        ER = Error register [HEX]
        ST = Status register [HEX]
Powered_Up_Time is measured from power on, and printed as
DDd+hh:mm:SS.sss where DD=days, hh=hours, mm=minutes,
SS=sec, and sss=millisec. It "wraps" after 49.710 days.

Error 2 occurred at disk power-on lifetime: 33029 hours (1376 days + 5 hours)
  When the command that caused the error occurred, the device was active or idle.

  After command completion occurred, registers were:
  ER ST SC SN CL CH DH
  -- -- -- -- -- -- --
  84 41 00 00 00 00 00  Error: ICRC, ABRT at LBA = 0x00000000 = 0

  Commands leading to the command that caused the error were:
  CR FR SC SN CL CH DH DC   Powered_Up_Time  Command/Feature_Name
  -- -- -- -- -- -- -- --  ----------------  --------------------
  60 00 20 80 be a6 40 08  23d+09:48:45.634  READ FPDMA QUEUED
  60 00 f0 00 bc a6 40 08  23d+09:48:45.631  READ FPDMA QUEUED
  60 00 c8 80 b9 a6 40 08  23d+09:48:45.628  READ FPDMA QUEUED
  60 00 a0 00 b7 a6 40 08  23d+09:48:45.626  READ FPDMA QUEUED
  60 00 78 80 b4 a6 40 08  23d+09:48:45.623  READ FPDMA QUEUED

Error 1 occurred at disk power-on lifetime: 12737 hours (530 days + 17 hours)
  When the command that caused the error occurred, the device was active or idle.

  After command completion occurred, registers were:
  ER ST SC SN CL CH DH
  -- -- -- -- -- -- --
  84 41 00 00 00 00 00  Error: ICRC, ABRT at LBA = 0x00000000 = 0

  Commands leading to the command that caused the error were:
  CR FR SC SN CL CH DH DC   Powered_Up_Time  Command/Feature_Name
  -- -- -- -- -- -- -- --  ----------------  --------------------
  60 80 a0 00 4e 55 40 08   4d+04:22:40.757  READ FPDMA QUEUED
  60 80 80 00 4c 55 40 08   4d+04:22:40.754  READ FPDMA QUEUED
  60 80 60 00 4a 55 40 08   4d+04:22:40.751  READ FPDMA QUEUED
  60 80 40 00 48 55 40 08   4d+04:22:40.749  READ FPDMA QUEUED
  60 80 20 00 46 55 40 08   4d+04:22:40.748  READ FPDMA QUEUED

SMART Self-test log structure revision number 1
Num  Test_Description    Status                  Remaining  LifeTime(hours)  LBA_of_first_error
# 1  Short offline       Completed without error       00%     46107         -
# 2  Extended offline    Completed without error       00%     46092         -
# 3  Short offline       Completed without error       00%     46018         -
# 4  Short offline       Completed without error       00%     45849         -
# 5  Extended offline    Completed without error       00%     45828         -
# 6  Short offline       Completed without error       00%     45681         -
# 7  Short offline       Completed without error       00%     45513         -
# 8  Short offline       Completed without error       00%     45345         -
# 9  Short offline       Completed without error       00%     45177         -
#10  Extended offline    Completed without error       00%     45084         -
#11  Short offline       Completed without error       00%     45009         -
#12  Short offline       Completed without error       00%     44841         -
#13  Short offline       Completed without error       00%     44673         -
#14  Short offline       Completed without error       00%     44505         -
#15  Extended offline    Completed without error       00%     44364         -
#16  Short offline       Completed without error       00%     44337         -
#17  Short offline       Completed without error       00%     44169         -
#18  Short offline       Completed without error       00%     44001         -
#19  Short offline       Completed without error       00%     43833         -
#20  Short offline       Completed without error       00%     43665         -
#21  Extended offline    Completed without error       00%     43620         -

SMART Selective self-test log data structure revision number 1
 SPAN  MIN_LBA  MAX_LBA  CURRENT_TEST_STATUS
    1        0        0  Not_testing
    2        0        0  Not_testing
    3        0        0  Not_testing
    4        0        0  Not_testing
    5        0        0  Not_testing
Selective self-test flags (0x0):
  After scanning selected spans, do NOT read-scan remainder of disk.
If Selective self-test is pending on power-up, resume after 0 minute delay.
```

you can see the short smart test performed here at operating hours 46,107. 

```
# 1  Short offline       Completed without error       00%     46107         -
```

and the same results show the drive power on hours as 46,107 hours.

```
9 Power_On_Hours                                                   0x0012   094   094   000    Old_age   Always       -       46107
```


based on this i do NOT think it is an issue with the script, i actually think it is an error with Synology DSM's storage manager.......

**Comment by 007revad** [comment]
It does seem like a bug in storage manager bug... which they won't fix if they're planning on removing SMART tests.

I'd fix storage manager myself if it wasn't 42,000 lines of javascript.

**Comment by wallacebrf** [comment]
Damn that's a lot of jana script
